taskipy/task_runner.py

```diff
@@ -93,14 +97,36 @@ def send_signal_to_task_process(signum: int, _frame) -> None:
 
         return process.returncode
 
-    def __pre_task(self, task_name: str):
-        return self.__find_hooks('pre', task_name)
+    def __resolve_variables(self, command: str) -> str:
```

Comment:
Maybe this should be separated into two methods?
1. `__resolve_variables`
2. `__format_command_with_variables`

Why? Because we only need to resolve the variables once. That said, we might have three commands to run (`pre`, `main`, `post`). In that case, we will resolve variables three times instead of just once.

If we split this method into two, we run the following code:

Pseudo:
```python
vars = resolve_variables()
...
pre = format_command('pre_{cmd}', vars)
main = format_command(cmd, vars)
post = format_command('post_{cmd}', vars)
```

Reply:
:+1: Sounds like a better idea to me. I'll look into breaking that apart tomorrow

Reply:
Currently, a task can have `use_vars` on it or it can be a global setting. If we want to resolve the variables before hand once, I assume we would need to look at the project setting, pre_task, task, and post_task to see if any of them are set to use variables. If so, then do the resolving, save the variables, pass it to a method to format the commands, and then pass those formatted commands down to the `__run_command_and_return_exit_code` method.